migration/migrator/migrations/course/20190528164528_numeric_id.py

```diff
@@ -0,0 +1,34 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("ALTER TABLE IF EXISTS users ADD COLUMN user_numeric_id varchar;")
+    database.execute("CREATE INDEX users_user_numeric_id_idx ON users using btree (user_numeric_id ASC NULLS LAST);")
+
+def down(config, database, semester, course):
+    """
+    Run down migration (rollback).
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    database.execute("DROP INDEX IF EXSISTS users_user_numeric_id_idx;")
+    database.execute("ALTER TABLE IF EXISTS users DROP COLUMN IF EXISTS user_numeric_id;")
```

Comment:
According to the [doc](https://submitty.org/developer/migrations):

> In general, your down migration should only undo the necessary changes to make earlier versions of the Submitty software work. For example, if your migration is adding a column to an existing table in the database, it is probably not necessary or desirable to delete that data in the down function. We don’t want to lose any data since we likely plan to return to this state in the future.

It would be sufficient to `DROP INDEX` only.

Reply:
The column is the real essential portion, but the index has its importance too.  It strikes me as pointless to rollback the index, but never the column.

Just as a side note — the kind of "data loss" on rolling back / dropping the column (in this migration) would be non-essential to any given course.

Reply:
Change made.  Parallel change also made to master DB migration (or else there would be data sync'ing problems to course DBs).